inflate.c

```diff
@@ -1120,6 +1120,8 @@ int32_t Z_EXPORT PREFIX(inflate)(PREFIX3(stream) *strm, int32_t flush) {
             state->mode = MEM;
             return Z_MEM_ERROR;
         }
+    } else if (INFLATE_NEED_CHECKSUM(strm) && !state->wsize && flush == Z_FINISH) {
+        inf_chksum(strm, put - (out - strm->avail_out), (out - strm->avail_out));
```

Comment:
Using same expression twice hurts readability... even if using temporary variable means adding one more line, this is still quite short delta.

Reply:
I'm not entirely sure what you mean, none of the conditions being checked here overlap.  It just so happens that NEED_UPDATEWINDOW and NEED_CHECKSUM are the same in most cases, but they wouldn't necessarily have to be.  And inf_chksum does something very update from update window.

Reply:
Are you referring to the pointer arithmetic?  I guess that can be baked out.  One second.

Reply:
I think that's what you mean, let me know if not.  Resolving for now.

Reply:
I meant the `(out - strm->avail_out)` part, which you did fix.... I know compiler is smart enough to do duplicate code elimination even on lower optimisation levels, but we really shouldn't rely on it.